src/pipeline.js

```diff
@@ -87,6 +87,7 @@ class Pipeline {
    * @returns {Pipeline} this
    */
   before(f) {
+    Pipeline.describe(f);
```

Comment:
might not be so performant to capture the stack trace for every function. since it is only used for `loglevel === 'silly'`, you might check the loglevel here.

Reply:
Makes sense.